File: src/src/com/greatguide/backend/route/Direction.java
package com.greatguide.backend.route;

public class Direction {
	public enum Directions
	{
		North,
		NorthEast,
		East,
		SouthEast,
		South,
		SouthWest,
		West,
		NorthWest
	}

	public static String ToShortString(int aDirection)
	{
		return ToShortString(aDirection);
	}

	public static String ToShortString(Directions direction)
	{
		switch (direction)
		{
		case North:
			return "N";
		case NorthEast:
			return "NE";
		case East:
			return "E";
		case SouthEast:
			return "SE";
		case South:
			return "S";
		case SouthWest:
			return "SW";
		case West:
			return "W";
		case NorthWest:
			return "NW";
		default:
			return "";
		}
	}

	public static Directions IntToDirection(int aDirection)
	{
		return Directions.values()[aDirection];
	}
	
	public static Directions FromHeading(float aHeading)
	{
		aHeading = (short)(aHeading % 360);
		if ((aHeading >= 22.5) && (aHeading < 67.5))
			return Directions.NorthEast;
		if ((aHeading >= 67.5) && (aHeading < 112.5))
			return Directions.East;
		if ((aHeading >= 112.5) && (aHeading < 157.5))
			return Directions.SouthEast;
		if ((aHeading >= 157.5) && (aHeading < 202.5))
			return Directions.South;
		if ((aHeading >= 202.5) && (aHeading < 247.5))
			return Directions.SouthWest;
		if ((aHeading >= 247.5) && (aHeading < 292.5))
			return Directions.West;
		if ((aHeading >= 292.5) && (aHeading < 337.5))
			return Directions.NorthWest;

		return Directions.North;
	}
}

File: old/Types.WM5/Direction.cs
using System;

namespace Nucleo.GoodGuide.Types
{
  public class Direction
  {
    public enum Directions
    {
      North = 0,
      NorthEast,
      East,
      SouthEast,
      South,
      SouthWest,
      West,
      NorthWest
    }

    public static string ToShortString(Int16 direction)
    {
      return ToShortString((Directions)direction);
    }
    public static string ToShortString(Directions direction)
    {
      switch (direction)
      {
        case Directions.North:
          return "N";
        case Directions.NorthEast:
          return "NE";
        case Directions.East:
          return "E";
        case Directions.SouthEast:
          return "SE";
        case Directions.South:
          return "S";
        case Directions.SouthWest:
          return "SW";
        case Directions.West:
          return "W";
        case Directions.NorthWest:
          return "NW";
        default:
          return "";
      }
    }
    public static Directions IntToDirection(Int16 direction)
    {
      return (Directions)direction;
    }
    public static Int16 DirectionToInt(Directions direction)
    {
      return (Int16)direction;
    }

    public static Directions FromHeading(Int16 heading)
    {
      heading = (short)(heading % 360);
      if ((heading >= 22.5) && (heading < 67.5))
        return Directions.NorthEast;
      if ((heading >= 67.5) && (heading < 112.5))
        return Directions.East;
      if ((heading >= 112.5) && (heading < 157.5))
        return Directions.SouthEast;
      if ((heading >= 157.5) && (heading < 202.5))
        return Directions.South;
      if ((heading >= 202.5) && (heading < 247.5))
        return Directions.SouthWest;
      if ((heading >= 247.5) && (heading < 292.5))
        return Directions.West;
      if ((heading >= 292.5) && (heading < 337.5))
        return Directions.NorthWest;

      return Directions.North;
    }
  }
}
